### PR TITLE
Fix misaligned pffft

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Resolves: #5906

Alternative fix for misalignment of buffers passed to pffft uses a non-default
allocator in std::vector.

Also PffftFloats is a pointer, always to the start of aligned span of floats (but agnostic
about the memory management).

Also moves FFT related files out of lib-math.

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

**Comment by Paul-Licameli** [comment]
There was some discussion in another channel.  @crsib give an opinion.

@saintmatthieu doesn't trust this because the allocation and destruction functions weren't hit in the debugger.  But what I wrote was not what was compiled in his build.

For unknown reasons my code compiles in CI for all Windows build but did not compile on Matt's machine.

**Review by crsib — APPROVED** [review]
I don't see anything wrong with this code and I don't want to be picky about the names

**Comment by saintmatthieu** [comment]
No I do get the allocators hit.
I can't build in `Debug`, though, compilation errors. @crsib

**Comment by saintmatthieu** [comment]
> Now I do get the allocators hit. I can't build in `Debug`, though, compilation errors, have to use `RelWithDebugInfo`. Is the debugging experience really as good as in `Debug` ? @crsib

Paul fixed this.

**Comment by Paul-Licameli** [comment]
I think a redefined allocator and a a neater solution, but there is more to it than a redefined deleted for unique_ptr.  I’m less familiar with this technique and muffed my first attempt a little!

File: libraries/lib-music-information-retrieval/PowerSpectrumGetter.h [hunk 1]
```diff
@@ -50,9 +51,154 @@ template<typename T> struct PffftAllocator
    }
 };
 
+//! An unsigned number of floats that can span two addresses that are well
+//! aligned for pffft
+struct PffftAlignedCount
+{
+   // see source for pffft_aligned_malloc
+   static constexpr size_t ByteAlignment = 64;
```
Comment (crsib) [inline libraries/lib-music-information-retrieval/PowerSpectrumGetter.h]: Is this constant valid, though? The library aligns memory regions to 64 bytes, but this is not required by the instruction sets it uses internally (SSE2/Neon).

Probably, it doesn't harm, so I won't block this PR for this reason.

**Review by crsib — APPROVED** [review]

**Comment by Paul-Licameli** [comment]
> Is this constant valid, though? The library aligns memory regions to 64 bytes, but this is not required by the instruction sets it uses internally (SSE2/Neon).
> 
> Probably, it doesn't harm, so I won't block this PR for this reason.

Yes, that is the weak point, but at least the magic number 64 is in only once place, reproducing a detail hidden in the pffft sources.

**Comment by Paul-Licameli** [comment]
Alternative fix for misalignment of buffers passed to pffft uses a non-default
allocator in std::vector.

Also PffftFloats is a pointer, always to the start of aligned span of floats (but agnostic
about the memory management).

Also moves FFT related files out of lib-math.

**Pull request merged**